common/include/vecmat.h

```diff
@@ -163,7 +163,7 @@ class vm_distance_squared
 	}
 	static constexpr vm_distance_squared maximum_value()
 	{
-		return vm_distance_squared{0x7fffffffffffffff};
+		return vm_distance_squared{static_cast<fix64>(0x7fffffffffffffff)};
```

Comment:
[Use stdint constants for some INT*_MAX](/dxx-rebirth/dxx-rebirth/commit/08e4a6e6203bec1b44326e6e7d1081a21b8f140a) changes this site as shown above and cleans up a few other places that integer literals had been hardcoded.  Based on your prior feedback, I expect this should resolve the clang failure you saw without introducing a gcc failure.